# Genetic-Sequence-Phylogeny
This repository contains a Python program that implements the Neighbor Joining algorithm to construct a phylogenetic tree based on genetic sequences. The program reads input sequences, calculates differences, and generates a phylogenetic tree representation. It provides functionality to create difference matrices, write output files, and visualize the tree. Explore evolutionary relationships among genetic sequences using this tool.

# Output
The edge file:

Provides a tabular representation of the relationships, where each line represents an edge. The parent and child nodes are specified by their IDs, and the distance between them is provided as a numerical value.

The genetic distances file:

Genetic distance, also known as evolutionary distance or genetic divergence, is a measure of the genetic differences or dissimilarities between two or more sequences or organisms. It quantifies the degree of genetic change that has occurred over time or through evolutionary processes.
the genetic distance is calculated as the proportion of mismatches between two genetic sequences divided by the length of the sequences. It provides a quantitative measure of how much the sequences differ from each other at the nucleotide or genetic level.

The genetic distance matrix generated by the program represents the genetic distances between all pairs of sequences in the input dataset. Each entry in the matrix represents the genetic distance between a pair of sequences.

tree.tre file:

The "tree.tre" file contains a textual representation of the phylogenetic tree constructed using the Neighbor Joining algorithm. The tree is represented in a hierarchical structure, where each node in the tree is enclosed in parentheses.
The structure of the Newick representation follows a set of conventions. Each node is labeled with an ID or a name, and the relationships between nodes are represented by nested parentheses. The branches or edges connecting the nodes are represented by colons followed by the distance or branch length.

This file can be visualized using an online tool such as the etoolkit viewer: http://etetoolkit.org/treeview/.

The example2 tree is visualized as:
![image](https://github.com/RLimond/Genetic-Sequence-Phylogeny/assets/129456338/0f1632a3-aa63-440a-b14e-301b3c2c7981)

